spec/duration.html

```diff
@@ -387,6 +387,41 @@ <h1>Temporal.Duration.prototype.minus ( _other_ [ , _options_ ] )</h1>
       </emu-alg>
     </emu-clause>
 
+    <emu-clause id="sec-temporal.duration.prototype.round">
+      <h1>Temporal.Duration.prototype.round ( _options_ )</h1>
+      <p>
+        The `round` method takes one argument _options_.
+        The following steps are taken:
+      </p>
+      <emu-alg>
+        1. Let _duration_ be the *this* value.
+        1. Perform ? RequireInternalSlot(_duration_, [[InitializedTemporalDuration]]).
+        1. Set _options_ to ? NormalizeOptionsObject(_options_).
+        1. Let _defaultLargestUnit_ be *"nanoseconds"*.
+        1. If _duration_.[[Years]] is not zero, set _defaultLargestUnit_ to *"years"*.
+        1. If _duration_.[[Months]] is not zero, set _defaultLargestUnit_ to *"months"*.
+        1. If _duration_.[[Weeks]] is not zero, set _defaultLargestUnit_ to *"weeks"*.
+        1. If _duration_.[[Days]] is not zero, set _defaultLargestUnit_ to *"days"*.
+        1. If _duration_.[[Hours]] is not zero, set _defaultLargestUnit_ to *"hours"*.
+        1. If _duration_.[[Minutes]] is not zero, set _defaultLargestUnit_ to *"minutes"*.
+        1. If _duration_.[[Seconds]] is not zero, set _defaultLargestUnit_ to *"seconds"*.
+        1. If _duration_.[[Milliseconds]] is not zero, set _defaultLargestUnit_ to *"milliseconds"*.
+        1. If _duration_.[[Microseconds]] is not zero, set _defaultLargestUnit_ to *"microseconds"*.
```

Comment:
Perhaps this could be refactored into an abstract op? I know it's only used here but it is a pure function of `duration` which looks messy inline.

Reply:
I agree, a named operation would make it clearer what the purpose is.